src/studio/ui/CellRenderer.java

```diff
@@ -48,25 +48,27 @@ public Component getTableCellRendererComponent(JTable table,
                                                    boolean hasFocus,
                                                    int row,
                                                    int column) {
-        K.KBase kb = (K.KBase) value;
-        String text = kb.toString(kb instanceof K.KBaseVector ? formatContextWithType : formatContextNoType);
-        text = Util.limitString(text, Config.getInstance().getMaxCharsInTableCell());
-        setText(text);
-        setForeground(kb.isNull() ? nullColor : fgColor);
+        if (value != null) {

```

Comment:
Have you really got NPE? I wonder how to reproduce. Overall I agree with the change. But I am curious as I am using the Studio a lot and don't remember of NPE...

Reply:
NPE was showing in the logs while i did some simple test. I was showing a table like ([] a:5#1e9) and click on the cell led to the NPE (only in logs)

Reply:
If you can capture the stack trace when the value gets null, it would be useful to my education. I tried few Java versions, and can't get null. Obviously it could depends on JRE version, operation system and selected L&F.

Reply:
```
21:15:22.198 [AWT-EventQueue-1] ERROR stderr - Exception in thread "AWT-EventQueue-1" java.lang.NullPointerException
21:15:22.199 [AWT-EventQueue-1] ERROR stderr - 	at studio.ui.CellRenderer.getTableCellRendererComponent(CellRenderer.java:55)
21:15:22.199 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.JTable$AccessibleJTable.getAccessibleChild(JTable.java:7037)
21:15:22.199 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.JTable$AccessibleJTable.getAccessibleAt(JTable.java:7424)
21:15:22.199 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.JTable$AccessibleJTable.valueChanged(JTable.java:6939)
21:15:22.200 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:184)
21:15:22.200 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.DefaultListSelectionModel.fireValueChanged(DefaultListSelectionModel.java:154)
21:15:22.200 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.DefaultListSelectionModel.setValueIsAdjusting(DefaultListSelectionModel.java:685)
21:15:22.200 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.plaf.basic.BasicTableUI$Handler.setValueIsAdjusting(BasicTableUI.java:953)
21:15:22.201 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.plaf.basic.BasicTableUI$Handler.mouseReleased(BasicTableUI.java:1166)
21:15:22.201 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
21:15:22.201 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Component.processMouseEvent(Component.java:6539)
21:15:22.201 [AWT-EventQueue-1] ERROR stderr - 	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
21:15:22.202 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Component.processEvent(Component.java:6304)
21:15:22.202 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Container.processEvent(Container.java:2239)
21:15:22.202 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Component.dispatchEventImpl(Component.java:4889)
21:15:22.202 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Container.dispatchEventImpl(Container.java:2297)
21:15:22.202 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Component.dispatchEvent(Component.java:4711)
21:15:22.203 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
21:15:22.203 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
21:15:22.203 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
21:15:22.203 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Container.dispatchEventImpl(Container.java:2283)
21:15:22.203 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Window.dispatchEventImpl(Window.java:2746)
21:15:22.204 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.Component.dispatchEvent(Component.java:4711)
21:15:22.204 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
21:15:22.204 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue.access$500(EventQueue.java:97)
21:15:22.204 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue$3.run(EventQueue.java:709)
21:15:22.204 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue$3.run(EventQueue.java:703)
21:15:22.205 [AWT-EventQueue-1] ERROR stderr - 	at java.security.AccessController.doPrivileged(Native Method)
21:15:22.205 [AWT-EventQueue-1] ERROR stderr - 	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
21:15:22.205 [AWT-EventQueue-1] ERROR stderr - 	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
21:15:22.205 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue$4.run(EventQueue.java:733)
21:15:22.205 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue$4.run(EventQueue.java:731)
21:15:22.206 [AWT-EventQueue-1] ERROR stderr - 	at java.security.AccessController.doPrivileged(Native Method)
21:15:22.206 [AWT-EventQueue-1] ERROR stderr - 	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
21:15:22.206 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
21:15:22.206 [AWT-EventQueue-1] ERROR stderr - 	at org.GNOME.Accessibility.AtkWrapper$6.dispatchEvent(AtkWrapper.java:717)
21:15:22.206 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
21:15:22.207 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
21:15:22.207 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
21:15:22.207 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
21:15:22.207 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
21:15:22.208 [AWT-EventQueue-1] ERROR stderr - 	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Reply:
Basically the null is given in JTable.getAccessibleChild(int i):
```java
public Accessible getAccessibleChild(int i) {
            if (i < 0 || i >= getAccessibleChildrenCount()) {
                return null;
            } else {
                // children increase across, and then down, for tables
                // (arbitrary decision)
                int column = getAccessibleColumnAtIndex(i);
                int row = getAccessibleRowAtIndex(i);

                TableColumn aColumn = getColumnModel().getColumn(column);
                TableCellRenderer renderer = aColumn.getCellRenderer();
                if (renderer == null) {
                    Class<?> columnClass = getColumnClass(column);
                    renderer = getDefaultRenderer(columnClass);
                }
                Component component = renderer.getTableCellRendererComponent(
                                  JTable.this, null, false, false,
                                  row, column);
                return new AccessibleJTableCell(JTable.this, row, column,
                      getAccessibleIndexAt(row, column));
            }
        }
```